### PR TITLE
port EF light client tests

### PR DESCRIPTION
**Pull request by etan-status**
EF has a number of unit tests related to light client functionality, see
https://github.com/ethereum/consensus-specs/blob/v1.1.0/tests/core/pyspec/eth2spec/test/altair/unittests/test_sync_protocol.py
The tests have been ported over and integrated into our test suite.

### CODE REVIEW

File: tests/official/altair/test_fixture_sync_protocol.nim [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  stew/bitops2,
+  # Beacon chain internals
+  ../../../beacon_chain/spec/
+    [forks, helpers, light_client_sync, signatures, state_transition],
+  # Mock helpers
+  ../../mocking/[mock_blocks, mock_genesis, mock_validator_keys],
+  # Test utilities
+  ../../testutil, ../../testblockutil
+
+proc payload =
```
Comment (arnetheduck) [inline tests/official/altair/test_fixture_sync_protocol.nim]: what's this proc for?

Comment (etan-status) [reply tests/official/altair/test_fixture_sync_protocol.nim]: It's a remnant from `test_process_justification_and_finalization`. I'm still investigating some other build issue, but the `payload` wrapper will be gone by the time this PR is through 'draft'.

**Comment by etan-status** [comment]
Depends on https://github.com/status-im/nimbus-eth2/pull/2928

**Comment by etan-status** [comment]
Still investigating the weird build issue on `linux-i386`

```
[Suite] Ethereum Foundation - Altair - Unittests - Sync protocol [Preset: mainnet]
787
    Unhandled exception: value out of range: -8765042107931517952 notin -2147483648 .. 2147483647 [RangeError]
788
  [FAILED] process_light_client_update_not_updated
789
    Unhandled exception: value out of range: -3840215400646785024 notin -2147483648 .. 2147483647 [RangeError]
790
  [FAILED] process_light_client_update_timeout
791
    Unhandled exception: value out of range: 3514154972565851136 notin -2147483648 .. 2147483647 [RangeError]
792
  [FAILED] process_light_client_update_finality_updated
793
```

**Comment by arnetheduck** [comment]
> Still investigating the weird build issue on linux-i386

this is usually an `int` being used in the wrong place - on i386, `int` is 32-bit while it's 64-bit elsewhere

**Comment by etan-status** [comment]
It seems to be just more of the same mock vs testblock shenanigans :-)

Depends on https://github.com/status-im/nimbus-eth2/pull/2936

**Comment by etan-status** [comment]
Finally green :-)

**Review by arnetheduck — APPROVED** [review]

**Pull request merged**